sys-apps/diskus/diskus-0.4.0.recipe

```diff
@@ -0,0 +1,240 @@
+
+SOURCE_URI_17="https://static.crates.io/crates/fuchsia-zircon-sys/fuchsia-zircon-sys-0.3.3.crate"
+CHECKSUM_SHA256_17="3dcaa9ae7725d12cdb85b3ad99a434db70b468c09ded17e012d86b5c1010f7a7"
+
+SOURCE_URI_18="https://static.crates.io/crates/humansize/humansize-1.1.0.crate"
+CHECKSUM_SHA256_18="b6cab2627acfc432780848602f3f558f7e9dd427352224b0d9324025796d2a5e"
+
+SOURCE_URI_19="https://static.crates.io/crates/kernel32-sys/kernel32-sys-0.2.2.crate"
+CHECKSUM_SHA256_19="7507624b29483431c0ba2d82aece8ca6cdba9382bff4ddd0f7490560c056098d"
+
+SOURCE_URI_20="https://static.crates.io/crates/lazy_static/lazy_static-1.1.0.crate"
+CHECKSUM_SHA256_20="ca488b89a5657b0a2ecd45b95609b3e848cf1755da332a0da46e2b2b1cb371a7"
+
+SOURCE_URI_21="https://static.crates.io/crates/libc/libc-0.2.43.crate"
+CHECKSUM_SHA256_21="76e3a3ef172f1a0b9a9ff0dd1491ae5e6c948b94479a3021819ba7d860c8645d"
+
+SOURCE_URI_22="https://static.crates.io/crates/lock_api/lock_api-0.1.4.crate"
+CHECKSUM_SHA256_22="775751a3e69bde4df9b38dd00a1b5d6ac13791e4223d4a0506577f0dd27cfb7a"
+
+SOURCE_URI_23="https://static.crates.io/crates/memoffset/memoffset-0.2.1.crate"
+CHECKSUM_SHA256_23="0f9dc261e2b62d7a622bf416ea3c5245cdd5d9a7fcc428c0d06804dfce1775b3"
+
+SOURCE_URI_24="https://static.crates.io/crates/nodrop/nodrop-0.1.12.crate"
+CHECKSUM_SHA256_24="9a2228dca57108069a5262f2ed8bd2e82496d2e074a06d1ccc7ce1687b6ae0a2"
+
+SOURCE_URI_25="https://static.crates.io/crates/num_cpus/num_cpus-1.8.0.crate"
+CHECKSUM_SHA256_25="c51a3322e4bca9d212ad9a158a02abc6934d005490c054a2778df73a70aa0a30"
+
+SOURCE_URI_26="https://static.crates.io/crates/owning_ref/owning_ref-0.3.3.crate"
+CHECKSUM_SHA256_26="cdf84f41639e037b484f93433aa3897863b561ed65c6e59c7073d7c561710f37"
+
+SOURCE_URI_27="https://static.crates.io/crates/parking_lot/parking_lot-0.6.4.crate"
+CHECKSUM_SHA256_27="f0802bff09003b291ba756dc7e79313e51cc31667e94afbe847def490424cde5"
+
+SOURCE_URI_28="https://static.crates.io/crates/parking_lot_core/parking_lot_core-0.3.1.crate"
+CHECKSUM_SHA256_28="ad7f7e6ebdc79edff6fdcb87a55b620174f7a989e3eb31b65231f4af57f00b8c"
+
+SOURCE_URI_29="https://static.crates.io/crates/rand/rand-0.5.5.crate"
+CHECKSUM_SHA256_29="e464cd887e869cddcae8792a4ee31d23c7edd516700695608f5b98c67ee0131c"
+
+SOURCE_URI_30="https://static.crates.io/crates/rand_core/rand_core-0.2.2.crate"
+CHECKSUM_SHA256_30="1961a422c4d189dfb50ffa9320bf1f2a9bd54ecb92792fb9477f99a1045f3372"
+
+SOURCE_URI_31="https://static.crates.io/crates/rand_core/rand_core-0.3.0.crate"
+CHECKSUM_SHA256_31="0905b6b7079ec73b314d4c748701f6931eb79fd97c668caa3f1899b22b32c6db"
+
+SOURCE_URI_32="https://static.crates.io/crates/rayon/rayon-1.0.3.crate"
+CHECKSUM_SHA256_32="373814f27745b2686b350dd261bfd24576a6fb0e2c5919b3a2b6005f820b0473"
+
+SOURCE_URI_33="https://static.crates.io/crates/rayon-core/rayon-core-1.4.1.crate"
+CHECKSUM_SHA256_33="b055d1e92aba6877574d8fe604a63c8b5df60f60e5982bf7ccbb1338ea527356"
+
+SOURCE_URI_34="https://static.crates.io/crates/redox_syscall/redox_syscall-0.1.40.crate"
```

Comment:
They're automatically fetched when in `Cargo.lock`, but they can be easily patched out of that (and they were, originally; I removed the patch in the latest commit) since it's only included in the main application, not the dependencies, and `Cargo.toml` has a way to specify them for their respected operating systems.

The thing is that cargo seems to need to *find* them when defined there, even when they're not pulled in; I discussed this [here](
https://github.com/haikuports/haikuports/pull/3333#discussion_r234051126). Patching it out of the `Cargo.toml` of those that require it is also another option, but is potentially tedious unless its scripted, too. Looking at it again, the format isn't as tricky as I thought it would be, so I'll see if it can be done. Hopefully there's a solution I'm missing, though, but it seems unlikely as Gentoo includes them as dependencies.

Reply:
I've now scripted a good amount of the process and was able to patch them all out, although some manual actions were needed with the script's current state.